### PR TITLE
yarden/open-api-fixes

### PR DESCRIPTION
**Pull request by yardenE**
[AZUEG-8748](https://spotinst.atlassian.net/browse/AZUEG-8748)

### CODE REVIEW

File: api/services/elastigroup/azure/spotVms/schemas/elastigroupVm-base-scheduling-task.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  isEnabled:
+    type: string
+    description: |
+      A list of scheduling tasks.
+  cronExpression:
+    type: string
+    description: |
+      A valid cron expression.
+      The cron is running in UTC time zone and is in Unix cron format of Cron Expression Validator Script.
+      Only one of 'frequency' or 'cronExpression' should be used at a time.
```
Comment (idohas) [inline api/services/elastigroup/azure/spotVms/schemas/elastigroupVm-base-scheduling-task.yaml:15]: Can you define enum here for the allowed values?

File: api/services/elastigroup/azure/spotVms/schemas/createElastigroupSpotVMs.yaml [hunk 2]
```diff
@@ -1115,4 +1114,158 @@ properties:
...
+                          description: |
+                            This will increase the target capacity by the defined amount when the scheduling task is triggered.
+                          minimum: 1
+                          example: 1
+                - allOf:
+                    - $ref: "elastigroupVm-base-scheduling-task.yaml"
+                    - type: object
+                      title: Scale Down Scheduled Task
+                      description: |
+                        This is a scheduled task that will allow to remove VMs from the the group's target capacity by a specific count.
```
Comment (idohas) [inline api/services/elastigroup/azure/spotVms/schemas/createElastigroupSpotVMs.yaml:1186]: "the" is repeated twice

File: api/services/elastigroup/azure/spotVms/schemas/createElastigroupSpotVMs.yaml [hunk 3]
```diff
@@ -1115,4 +1114,158 @@ properties:
...
+                          description: |
+                            This will decrease the target capacity by the defined amount when the scheduling task is triggered.
+                          minimum: 1
+                          example: 1
+                - allOf:
+                    - $ref: "elastigroupVm-base-scheduling-task.yaml"
+                    - type: object
+                      title: Scale Up Percentage Scheduled Task
+                      description: |
+                        This is a scheduled task that will allow to add VMs to the the group's target capacity by a percentage of the groups current target.
```
Comment (idohas) [inline api/services/elastigroup/azure/spotVms/schemas/createElastigroupSpotVMs.yaml:1207]: 1. "the" is repeated twice,
2. "percentage of the group's..."

File: api/services/elastigroup/azure/spotVms/schemas/elastigroupVm-scheduling.yaml [hunk 4]
```diff
@@ -1,2 +1,155 @@
...
+                  description: |
+                    This will decrease the target capacity by the defined amount when the scheduling task is triggered.
+                  minimum: 1
+                  example: 1
+        - allOf:
+            - $ref: "elastigroupVm-base-scheduling-task.yaml"
+            - type: object
+              title: Scale Up Percentage Scheduled Task
+              description: |
+                This is a scheduled task that will allow to add VMs to the the group's target capacity by a percentage of the groups current target.
```
Comment (idohas) [inline api/services/elastigroup/azure/spotVms/schemas/elastigroupVm-scheduling.yaml:91]: "the" is repeated twice

File: api/services/elastigroup/azure/spotVms/schemas/createElastigroupSpotVMs.yaml [hunk 5]
```diff
@@ -1115,4 +1114,158 @@ properties:
...
+                            This will increase the target capacity by the defined percentage value when the scheduling task is triggered.
+                            This will set the target capacity by the defined percentage of the current target when the scheduling task triggers.
+                          minimum: 1
+                          maximum: 100
+                - allOf:
+                    - $ref: "elastigroupVm-base-scheduling-task.yaml"
+                    - type: object
+                      title: Scale Down Percentage Scheduled Task
+                      description: |
+                        This is a scheduled task that will allow to remove VMs from the the group's target capacity by a percentage of the groups current target.
```
Comment (idohas) [inline api/services/elastigroup/azure/spotVms/schemas/createElastigroupSpotVMs.yaml:1229]: 1. "the" is repeated twice
2. "percentage of the group's..."

File: api/services/elastigroup/azure/spotVms/schemas/createElastigroupSpotVMs.yaml [hunk 6]
```diff
@@ -1115,4 +1114,158 @@ properties:
...
+                          type: integer
+                          description: |
+                            This will set the defined maximum group capacity when the scheduling task is triggered.
+                          minimum: 0
+                - allOf:
+                    - $ref: "elastigroupVm-base-scheduling-task.yaml"
+                    - type: object
+                      title: Scale Up Scheduled Task
+                      description: |
+                        This is a scheduled task that will allow to add VMs to the the group's target capacity by a specific count.
```
Comment (idohas) [inline api/services/elastigroup/azure/spotVms/schemas/createElastigroupSpotVMs.yaml:1165]: "the" is repeated twice

File: api/services/elastigroup/azure/spotVms/schemas/elastigroupVm-scheduling.yaml [hunk 7]
```diff
@@ -1,2 +1,155 @@
...
+    items:
+      anyOf:
+        - allOf:
+            - $ref: "elastigroupVm-base-scheduling-task.yaml"
+            - type: object
+              title: Scaling Scheduled Task
+              description: |
+                This is a scheduled task that will allow to change the group's capacty. If the target
+                is changed, then VMs will either be scaled up or down.
+                Note: At least one of the following must be configured: [scaleTargetCapacity, scaleMaxCapacity, scaleMinCapacity]
```
[the remark this answers is not in the document]
Comment (yardenE) [reply api/services/elastigroup/azure/spotVms/schemas/elastigroupVm-scheduling.yaml:20]: This isn't an enumeration; this is 3 properties that at least one of them must be set.

File: api/services/elastigroup/azure/spotVms/schemas/elastigroupVm-base-scheduling-task.yaml (hunk 1 above)
Comment (yardenE) [reply api/services/elastigroup/azure/spotVms/schemas/elastigroupVm-base-scheduling-task.yaml:15]: This isn't an enumeration; this is 2 properties that at least one of them must be set.

**Review by idohas — APPROVED** [review]

**Pull request merged**